cvxpy/atoms/affine/binary_operators.py

```diff
@@ -454,6 +454,13 @@ def scalar_product(x, y):
     return cvxpy_sum(prod)
 
 
+def scalar_product(x, y):
+    x = deep_flatten(x)
+    y = deep_flatten(y)
+    prod = multiply(conj(x), y)
+    return cvxpy_sum(prod)
```

Comment:
Maybe this is simpler:

```suggestion
def scalar_product(x, y):
    """
    Alias for vdot.
    """
    return vdot(x, y)
```

Reply:
Yes I agree, thanks for the suggestion.